src/models/mSeriesChallenge.ts

```diff
@@ -0,0 +1,268 @@
+      return new MSeriesChallengeParticipant(mSeriesChallengeParticipant, this.sessionHandler)
+    } else {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:show', { mSeriesChallengeId: this._mSeriesChallengeData.id, userId: this.sessionHandler.userId }) as MSeriesChallengeParticipantResponse
+      const p = new MSeriesChallengeParticipant(mSeriesChallengeParticipant, this.sessionHandler)
+      this.setMSeriesChallengeParticipant(p)
+      return p
+    }
+  }
+
+  /**
+   * Method to search for participants in challenge.
+   * Session User must either own or have joined challenge to search for other participants.
+   *
+   * @param options.nameSearchQuery Name to search for
+   * @param options.sort default: 'name'
+   * @param options.ascending default: true
+   * @param options.limit default: 20
+   * @param options.offset default: 0
+   * @returns An array of MSeries Challenge Participants
+   */
+  async getParticipants (options: {
+    nameSearchQuery: string
+    sort?: MSeriesChallengeParticipantSorting
+    ascending?: boolean
+    limit?: number
+    offset?: number
+  }) {
+    const participants = await this.action('mSeriesChallengeParticipant:list', {
+      mSeriesChallengeId: this._mSeriesChallengeData.id,
+      userId: this.sessionHandler.userId,
+      ...options
+    }) as MSeriesChallengeParticipantListResponse
+    return new MSeriesChallengeParticipants(participants.mSeriesChallengeParticipants, participants.mSeriesChallengeParticipantsMeta, this.sessionHandler)
+  }
+
+  /**
+   * Method to get leaderboard of challenge.
+   * Session User must either own or have joined challenge to get leaderboard.
+   *
+   * @param options.targetUserId userId to "center" leaderboard on
+   * @param options.ascending default: true
+   * @param options.limit default: 20
+   * @param options.offset default: 0
+   * @returns An array of MSeries Challenge Participants sorted by rank
+   */
+  async getLeaderboard (options: {
+    targetUserId?: number
+    ascending?: boolean
+    limit?: number
+    offset?: number
+  } = {}) {
+    const leaderboard = await this.action('mSeriesChallenge:leaderboard', {
+      id: this._mSeriesChallengeData.id,
+      userId: this.sessionHandler.userId,
+      ...options
+    }) as MSeriesChallengeLeaderboardResponse
+
+    return new MSeriesChallengeLeaderboardParticipants(leaderboard.mSeriesChallengeParticipants, leaderboard.mSeriesChallengeParticipantsMeta, this.sessionHandler)
+  }
+
+  /**
+   * Method to update challenge meta data.
+   * Session User must own challenge to update it.
+   *
+   * @param options.name
+   * @param options.userLimit
+   */
+  async update (options: {
+    name?: string
+    // isPublic?: boolean
+    userLimit?: number
+  }) {
+    await this.action('mSeriesChallenge:update', {
+      id: this._mSeriesChallengeData.id,
+      ...options
+    }) as MSeriesChallengeResponse
+
+    await this.reload()
```

Comment:
You shouldn't need to call reload here.  The action should return the model you need and you should use the set function to set it.